zebra-consensus/src/block/subsidy/general.rs

```diff
@@ -0,0 +1,203 @@
+//! Block and Miner subsidies, halvings and target spacing modifiers. - [§7.7][7.7]
+//!
+//! [7.7]: https://zips.z.cash/protocol/protocol.pdf#subsidies
+
+use std::convert::TryFrom;
+
+use zebra_chain::{
+    amount::{Amount, Error, NonNegative},
+    block::Height,
+    parameters::{Network, NetworkUpgrade::*},
+};
+
+use super::founders_reward::founders_reward;
+use crate::parameters::subsidy::*;
+
+/// `SlowStartShift()` as described in [protocol specification §7.7][7.7]
+///
+/// [7.7]: https://zips.z.cash/protocol/protocol.pdf#subsidies
+fn slow_start_shift() -> Height {
+    Height(SLOW_START_INTERVAL.0 / 2)
+}
+
+/// The divisor used for halvings.
+///
+/// `1 << Halving(height)`, as described in [protocol specification §7.7][7.7]
+///
+/// [7.7]: https://zips.z.cash/protocol/protocol.pdf#subsidies
+pub fn halving_divisor(height: Height, network: Network) -> u64 {
+    if height < SLOW_START_INTERVAL {
+        panic!("can't verify before block {}", SLOW_START_INTERVAL.0)
+    }
```

Comment:
This check isn't in the `Halving` definition in the spec?
```suggestion
```

Reply:
Hmm, actually, I think this function will be easier to understand if it is refactored slightly.

Reply:
I made this change in 8ec0264a00c15c6ea80cf213a39e3a0fb5005dd8 in oxarbitrage#29